### PR TITLE
E2e deployment logic refactor

### PR DESCRIPTION
**Pull request by jsturtevant**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Does some clean up in the e2e specs. I was working on adding the ability to add Windows to the LB tests and noticed we had two ways to create deployments and it wasn't easy to tweak the deployment to add the necessary Windows components.  This was a way to consolidate some of that logic.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**TODOs**:

- [X] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/azure_lb.go [hunk 1]
```diff
@@ -204,33 +123,18 @@ func AzureLBSpec(ctx context.Context, inputGetter func() AzureLBSpecInput) {
 			Clientset: clientset,
 		}
 		WaitForJobComplete(context.TODO(), ilbJobInput, e2eConfig.GetIntervals(specName, "wait-job")...)
+
+		if !input.SkipCleanup {
+			By("deleting the elb test resources")
```
Comment (CecileRobertMichon) [inline test/e2e/azure_lb.go:128]: should this be `ilb` ?

**Review by CecileRobertMichon — COMMENTED** [review]
lgtm

/assign @mboersma

File: test/e2e/azure_lb.go (hunk 1 above)
Comment (jsturtevant) [reply test/e2e/azure_lb.go]: yes fixed!

File: Makefile [hunk 2]
```diff
@@ -173,7 +173,7 @@ test-e2e-local: $(ENVSUBST) $(KUBECTL) $(GINKGO) ## Run e2e tests
 	PULL_POLICY=IfNotPresent $(MAKE) docker-build
 	MANAGER_IMAGE=$(CONTROLLER_IMG)-$(ARCH):$(TAG) \
 	$(ENVSUBST) < $(E2E_CONF_FILE) > $(E2E_CONF_FILE_ENVSUBST) && \
-	$(GINKGO) -v -trace -tags=e2e -focus="$(GINKGO_FOCUS)" -skip="${GINKGO_SKIP}" -nodes=$(GINKGO_NODES) --noColor=$(GINKGO_NOCOLOR) ./test/e2e -- \
+	$(GINKGO) -v -trace -tags=e2e -focus="$(GINKGO_FOCUS)" -skip="${GINKGO_SKIP}" -nodes=$(GINKGO_NODES) --noColor=$(GINKGO_NOCOLOR) -stream --progress ./test/e2e -- \
```
Comment (mboersma) [inline Makefile:176]: I often make this change (adding `-stream -progress`) locally, so :+1:.

**Review by mboersma — APPROVED** [review]
lgtm

This is good refactoring--should be more readable and maintainable. 💚

**Review by devigned — APPROVED** [review]
/lgtm
/approve

File: Makefile (hunk 2 above)
Comment (nader-ziada) [reply Makefile:176]: me too :D

Comment (CecileRobertMichon) [reply Makefile:176]: I will make sure I include this in #1018 when I rebase :)

**Pull request merged**